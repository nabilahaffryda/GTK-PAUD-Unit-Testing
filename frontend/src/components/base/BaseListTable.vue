<template>
  <v-data-table
    :hide-default-footer="!usePaging"
    :items="item || []"
    :options.sync="options"
    :items-per-page="limit"
    :server-items-length="total"
    :loading="loading"
    disable-sort
    :hide-default-header="hideHeader"
    :headers="headers"
    :footer-props="{ itemsPerPageOptions: [limit] }"
  >
    <template v-slot:header="{ props: { headers } }">
      <thead>
        <slot name="header" :headers="headers"></slot>
      </thead>
    </template>
    <template v-slot:body="{ items }">
      <tbody>
        <template v-if="items.length">
          <tr v-for="(item, i) in items" :key="i" class="transparent">
            <slot :item="item" :index="i" />
          </tr>
        </template>
        <template v-else>
          <tr>
            <td :colspan="headers.length">
              {{
                keyword
                  ? `Data Pencarian ${title}${keyword ? ` dengan kata kunci ${keyword}` : ''} tidak ditemukan`
                  : 'Belum ada data yang ditemukan'
              }}
            </td>
          </tr>
        </template>
      </tbody>
    </template>
  </v-data-table>
</template>

<script>
export default {
  name: 'BaseListTable',
  props: {
    usePaging: {
      type: Boolean,
      default: true,
    },
    loading: {
      type: Boolean,
      default: true,
    },
    hideHeader: {
      type: Boolean,
      default: false,
    },
    headers: {
      type: Array,
      default: () => [],
    },
    item: {
      type: Array,
      default: () => [],
    },
    opt: {
      type: Object,
      default: () => {},
    },
    total: {
      type: Number,
      default: 0,
    },
    limit: {
      type: Number,
      default: 10,
    },
    title: {
      type: String,
      default: '',
    },
    keyword: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      options: this.opt || {},
    };
  },
  methods: {
    fetch() {
      this.$emit('fetch', {
        page: this.options.page,
        limit: this.options.itemsPerPage,
      });
    },
  },
  watch: {
    options: {
      handler() {
        this.fetch();
      },
      deep: true,
    },
  },
};
</script>

<style scoped></style>
