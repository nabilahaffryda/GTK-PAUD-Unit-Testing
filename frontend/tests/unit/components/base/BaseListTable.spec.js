import { mount, createLocalVue } from "@vue/test-utils";
import BaseListTable from "@/components/base/BaseListTable.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseListTable.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('should show data in list', () => {
        const VDataTable = {
            props: ['items', 'loading', 'usePaging', 'hideHeader',
                'showSelect', 'keyword', 'title', 'limit', 'opt', 'total', 'headers'],
            template: '<div><slot :item="items[0]" name="item.name" /></div>'
        }

        const wrapper = mount(BaseListTable, {
            stubs: {
                VDataTable
            },
            localVue,
            vuetify
        })
        expect(wrapper).toMatchSnapshot();
    })
})
