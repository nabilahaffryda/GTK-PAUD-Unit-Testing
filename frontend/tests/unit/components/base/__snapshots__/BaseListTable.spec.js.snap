// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseListTable.vue should show data in list 1`] = `<div options="[object Object]" items-per-page="10" server-items-length="0" disable-sort="" footer-props="[object Object]"></div>`;
